src/test/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/AnnotatedVariantProducerUnitTest.java

```diff
@@ -13,233 +15,163 @@
-    @Test(groups = "sv", dataProvider = "forIntegrativeTest")
-    public void testIntegrative(final TestDataForSimpleSVs testData,
-                                final List<String> expectedAttributeKeys,
-                                final Broadcast<SVIntervalTree<VariantContext>> broadcastCNVCalls,
-                                final Broadcast<ReferenceMultiSource> referenceBroadcast,
-                                final Broadcast<SAMSequenceDictionary> refSeqDictBroadcast) throws IOException {
-
-        final NovelAdjacencyAndAltHaplotype breakpoints = testData.biPathBubble;
-        final List<SimpleChimera> evidence =
-                Collections.singletonList(new SimpleChimera(testData.firstAlignment, testData.secondAlignment,
-                        Collections.emptyList(), testData.evidenceAssemblyContigName, NO_GOOD_MAPPING_TO_NON_CANONICAL_CHROMOSOME,
-                                SVDiscoveryTestUtilsAndCommonDataProvider.b37_seqDict_20_21));
-        final String sampleID = "testSample";
-
-        final VariantContext variantContext =
-                AnnotatedVariantProducer
-                        .produceAnnotatedVcFromInferredTypeAndRefLocations(
-                                breakpoints,
-                                DiscoverVariantsFromContigAlignmentsSAMSpark.inferSimpleTypeFromNovelAdjacency(breakpoints),
-                                evidence,
-                                referenceBroadcast, refSeqDictBroadcast, broadcastCNVCalls, sampleID);
-
-        final List<String> attributeKeys = variantContext.getAttributes().keySet().stream().sorted().collect(Collectors.toList());
-
-        Assert.assertEquals(attributeKeys, expectedAttributeKeys);
+    @Test(groups = "sv", dataProvider = "forAssemblyBasedAnnotation")
+    public void testAssemblyBasedAnnotation(final List<SvType> inferredTypes,
+                                            final SimpleNovelAdjacencyAndChimericAlignmentEvidence novelAdjacencyAndAssemblyEvidence,
+                                            final Broadcast<ReferenceMultiSource> broadcastReference,
+                                            final Broadcast<SAMSequenceDictionary> broadcastSequenceDictionary,
+                                            final String sampleId,
+                                            final String linkKey,
+                                            final List<VariantContext> expectedVariants) throws IOException {
+        if (inferredTypes.size() == 1) {
+            VariantContextTestUtils.assertVariantContextsAreEqual(produceAnnotatedVcFromAssemblyEvidence(inferredTypes.get(0), novelAdjacencyAndAssemblyEvidence, broadcastReference, broadcastSequenceDictionary, null, sampleId).make(),
+                    expectedVariants.get(0), Arrays.asList(CONTIG_NAMES, HQ_MAPPINGS)); // these two are omitted because: 1) HQ_MAPPINGS equal testing code is wrong (saying 1!=1), 2)CONTIG_NAMES will be tested in another test
```

Comment:
What do you mean by the first of these comments? Can we debug what's going on there?